rasa/core/policies/rule_policy.py

```diff
@@ -263,11 +413,18 @@ def train(
             trackers_as_states, trackers_as_actions
         )
 
-        # TODO use story_trackers and rule_trackers
-        #  to check that stories don't contradict rules
+        # make this configurable because checking might take a lot of time
+        if self._check_rules_with_stories:
+            # using trackers here might not be the most efficient way, however
+            # it allows us to directly test `predict_action_probabilities` method
+            self._find_contradicting_rules(training_trackers, domain, interpreter)
 
         logger.debug(f"Memorized '{len(self.lookup[RULES])}' unique rules.")
 
+    def _output_debug_logs(self, message: Text) -> None:
```

Comment:
I think it's cleaner and more robust to temporarily modify the log level of the module logger using `logger.setLevel(logging.WARNING)`

Reply:
👍 I forgot about it